vm_api/src/trace.rs

```diff
@@ -105,6 +114,28 @@ impl ExpectInvocation {
                 id, p, invoc.params
             );
         }
+
+        // match emitted events
+        let emitted_events = &invoc.events;
+        let expected_events = &self.events;
+        assert_eq!(
+            emitted_events.len(),
+            expected_events.len(),
+            "{} {} emitted={}, expected={}, {:?}, {:?}",
+            id,
+            "length of expected and emitted events do not match",
+            emitted_events.len(),
+            expected_events.len(),
+            emitted_events,
+            expected_events
+        );
+
+        for (i, event) in emitted_events.iter().enumerate() {
+            let expect_event = expected_events.get(i).unwrap();
```

Comment:
Use `zip()` to iterate the vectors in parallel

Reply:
Done.

Reply:
It looks like you forgot to delete this block of code after adding the zipped version above.